### ISSUE TITLE
Frequent crashes on Arch Linux

### ISSUE THREAD

**Issue by doughsay** [Alecaddd/sequeler#284]
I'm not sure what's going on or how to get more information, below is the only thing I get when launching from the command-line:
```
(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.618: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed

(Sequeler:265799): GLib-GObject-CRITICAL **: 18:19:52.619: g_value_get_long: assertion 'G_VALUE_HOLDS_LONG (value)' failed
fish: “com.github.alecaddd.sequeler” terminated by signal SIGSEGV (Address boundary error)
```

**Comment by Alecaddd** [comment]
I guess you're using via Flatpak.
This doesn't happen on elementaryOS, sorry. Please, open an issue on the flathub repo as I'm not the maintainer of that fork.
https://github.com/flathub/com.github.alecaddd.sequeler

**Comment by doughsay** [comment]
I did not use the flatpak, I built it from AUR, which builds directly from this git repo: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=sequeler-git

**Comment by Alecaddd** [comment]
Mhhh, interesting.
Which Database are you using?
Based on the terminal debug, it seems that the `get_long()` method is causing the segfault.
I'm using that method only once to fetch the rows count of each table: https://github.com/Alecaddd/sequeler/blob/a95c4ca557142f3a8a2d460b04bd5b4042273cb7/src/Layouts/DataBaseSchema.vala#L308

**Comment by doughsay** [comment]
I was viewing a Postgres 9.6 database when this happened.  If you want me to try some specific steps I can see if I can help.  Also: I haven't tried this again since I opened the issue, because I was evaluating several different linux db clients (I haven't found anything suitable yet).  If we can figure out the segfaults here then this would be my tool of choice!

So, I can try a few things again later since there have been a few commits since the last time I tried.
